lib/perl/centreon/script/centcore.pm

```diff
@@ -868,7 +868,7 @@ sub syncTraps($) {
             if (defined($ns_server->{remote_id}) && $ns_server->{remote_id} != 0 && $self->{instance_mode} ne "remote") {
                 $remote_server = $self->getServerConfig($ns_server->{remote_id});
                 $port = checkSSHPort($remote_server->{ssh_port});
-                $cmd = "$self->{scp} -r -P $port /etc/snmp/centreon_traps/$id/centreontrapd.sdb $remote_server->{'ns_ip_address'}:/etc/snmp/centreon_traps/";
+                $cmd = "$self->{scp} -r -P $port /etc/snmp/centreon_traps/$id $remote_server->{'ns_ip_address'}:/etc/snmp/centreon_traps/";
```

Comment:
please check/protect  $self->{scp} and $port and $remote_server->{'ne_ip_address'} against RCE flaws !

Reply:
$self->{scp] is hard coded and is never overwritten by something else
$port comes from centreon.nagios_server table, the column is int(11) 
$remote_server->{'ns_ip_address'} comes from the same table ns_ip_address is varchar(255) . There is one issue there. We do not filter all the special characters.  We let people use the hostname or the IP address. Each one being normalized through various RFC. 
https://en.wikipedia.org/wiki/Hostname

I was able to use ! and ?  meaning that people could alter the command line. In most cases it should only break the scp command line and nothing harmfull should be possible. But that doesn't mean it is impossible (I just don't see what could go very wrong there) 

Anyway, there is a security issue there, but it is not on the centcore side. We need to implement a new php class that makes sure the "ip address" field of the form is a valid IP address (meaning that it matches the RFC) or a valid hostname/FQDN (meaning that it matches the RFC). Perhaps there's already a function that does that in PHP.